include/heuristic/HeuristicMapper.hpp

```diff
@@ -93,13 +103,45 @@ class HeuristicMapper : public Mapper {
       } else {
         throw QMAPException("Something wrong in applySWAP.");
       }
+
+      if (considerFidelity) {
+        // accounting for fidelity difference of single qubit gates (two qubit
+        // gates are handled in the heuristic)
+        costFixed +=
+            (-(singleQubitGateMultiplicity.at(static_cast<std::size_t>(q2)) -
+               singleQubitGateMultiplicity.at(static_cast<std::size_t>(q1))) *
+                 log2(singleQubitFidelities.at(swap.first)) -
+             (singleQubitGateMultiplicity.at(static_cast<std::size_t>(q1)) -
+              singleQubitGateMultiplicity.at(static_cast<std::size_t>(q2))) *
+                 log2(singleQubitFidelities.at(swap.second)));
+        // adding cost oft the swap gate
+        if (arch.bidirectional()) {
+          costFixed +=
+              -3 * log2(twoQubitFidelities.at(swap.first).at(swap.second));
```

Comment:
Prefer to use `std::log2` instead of `log2` (which is, technically, not guaranteed to be defined).

Reply:
Thank you, will be fixed in the next commit.